mustang_core/lib/src/widgets/route_redirect.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/widgets.dart';
+
+class RouteRedirect extends StatelessWidget {
+  const RouteRedirect({
+    Key? key,
+    required this.test,
+    required this.targetRouteName,
+    required this.child,
+    this.pushReplace = true,
+    this.loadingMsg = 'Redirecting...',
+  }) : super(key: key);
+
+  /// If [test] returns true, this widget redirects the app to the [targetRoute],
+  /// otherwise noop
+  final bool Function() test;
+
+  /// Route to go to when [test] is true
+  final String targetRouteName;
+
+  /// If [pushReplace] is true, [targetRoute] replaces all existing routes,
+  /// otherwise [targetRoute] is pushed on top of the existing route
+  final bool pushReplace;
+
+  /// If [test] return false, child gets rendered
+  final Widget child;
+
+  /// Text to show while transitioning to the destination route
+  final String loadingMsg;
+
+  @override
+  Widget build(BuildContext context) {
+    if (test()) {
+      if (pushReplace) {
+        SchedulerBinding.instance?.addPostFrameCallback((_) async {
+          Navigator.pushNamedAndRemoveUntil(
+            context,
+            targetRouteName,
+            (route) => false,
+          );
+        });
```

Comment:
looking that the example...when the session will invalid the block below will get triggered in the code and then the user will be redirected to the login screen...
```
SchedulerBinding.instance?.addPostFrameCallback((_) async {
            await JobDetailsService().memoizedGetData();
          })
```
so in the background, the API call was made even when token is invalid

Reply:
I think the builder should be wrapped with this widget (in the example)...that way we can prevent making API call when we know the token is invalid